pkg/apis/azureprovider/v1alpha1/azureclusterproviderconfig_types.go

```diff
@@ -48,6 +48,9 @@ type AzureClusterProviderSpec struct {
 	// SAKeyPair is the service account key pair.
 	SAKeyPair KeyPair `json:"saKeyPair,omitempty"`
 
+	AdminKubeconfig string `json:"adminKubeconfig,omitempty"`
+
+	DiscoveryHashes []string `json:"discoveryHashes,omitempty"`
```

Comment:
Similar to my comment about `AdminKubeconfig`, if there is a need to be able to reference this without generating it on demand, consider moving this to Status.

Reply:
Thanks it makes sense, updated PR